src/app/fdctl/run/tiles/fd_replay.c

```diff
@@ -0,0 +1,731 @@
+  ctx->poh_out_mem    = topo->workspaces[ topo->objs[ poh_out_link->dcache_obj_id ].wksp_id ].wksp;
+  ctx->poh_out_chunk0 = fd_dcache_compact_chunk0( ctx->poh_out_mem, poh_out_link->dcache );
+  ctx->poh_out_wmark  = fd_dcache_compact_wmark( ctx->poh_out_mem, poh_out_link->dcache, poh_out_link->mtu );
+  ctx->poh_out_chunk  = ctx->poh_out_chunk0;
+
+/* Set up stake weights tile output */
+  fd_topo_link_t * stake_weights_out = &topo->links[ tile->out_link_id_primary ];
+  ctx->stake_weights_out_mcache = stake_weights_out->mcache;
+  ctx->stake_weights_out_sync   = fd_mcache_seq_laddr( ctx->stake_weights_out_mcache );
+  ctx->stake_weights_out_depth  = fd_mcache_depth( ctx->stake_weights_out_mcache );
+  ctx->stake_weights_out_seq    = fd_mcache_seq_query( ctx->stake_weights_out_sync );
+  ctx->stake_weights_out_mem    = topo->workspaces[ topo->objs[ stake_weights_out->dcache_obj_id ].wksp_id ].wksp;
+  ctx->stake_weights_out_chunk0 = fd_dcache_compact_chunk0( ctx->stake_weights_out_mem, stake_weights_out->dcache );
+  ctx->stake_weights_out_wmark  = fd_dcache_compact_wmark ( ctx->stake_weights_out_mem, stake_weights_out->dcache, stake_weights_out->mtu );
+  ctx->stake_weights_out_chunk  = ctx->stake_weights_out_chunk0;
+
+  /* Valloc setup */
+  void * alloc_shalloc = fd_alloc_new( alloc_shmem, 3UL );
+  if( FD_UNLIKELY( !alloc_shalloc ) ) { 
+    FD_LOG_ERR( ( "fd_allow_new failed" ) ); }
+  fd_alloc_t * alloc = fd_alloc_join( alloc_shalloc, 3UL );
+  if( FD_UNLIKELY( !alloc ) ) {
+    FD_LOG_ERR( ( "fd_alloc_join failed" ) ); 
+  }
+
+  char hostname[64];
+  gethostname(hostname, sizeof(hostname));
+  ulong hashseed = fd_hash(0, hostname, strnlen(hostname, sizeof(hostname)));
+
+  // Allocate new wksp
+  fd_wksp_t * funk_wksp = fd_wksp_new_anonymous( FD_SHMEM_GIGANTIC_PAGE_SZ, 256UL, 0UL, "wksp", 0UL );
+  if (funk_wksp == NULL)
+    FD_LOG_ERR(( "failed to attach to workspace" ));
+  fd_wksp_reset( funk_wksp, (uint)hashseed);
+
+  // TODO: is this required here?
+  /* Create scratch region */
+  ulong  smax   = 1024UL /*MiB*/ << 21;
+  ulong  sdepth = 128;      /* 128 scratch frames */
+  void * smem   = fd_wksp_alloc_laddr( ctx->wksp, fd_scratch_smem_align(), fd_scratch_smem_footprint( smax   ), 421UL );
+  void * fmem   = fd_wksp_alloc_laddr( ctx->wksp, fd_scratch_fmem_align(), fd_scratch_fmem_footprint( sdepth ), 421UL );
+  FD_TEST( (!!smem) & (!!fmem) );
+  fd_scratch_attach( smem, fmem, smax, sdepth );
+
+  // Funky setup
+  fd_funk_t * funk;
+  void * shmem;
+  shmem = fd_wksp_alloc_laddr( funk_wksp, fd_funk_align(), fd_funk_footprint(), FD_FUNK_MAGIC );
+  if (shmem == NULL)
+    FD_LOG_ERR(( "failed to allocate a funky" ));
+  funk = fd_funk_join(fd_funk_new(shmem, 42, hashseed, 1024UL, 300000000UL));
+  if (funk == NULL) {
+    fd_wksp_free_laddr(shmem);
+    FD_LOG_ERR(( "failed to allocate a funky" ));
+  }
+
+  inner_valloc = fd_alloc_virtual( alloc );
+  fd_valloc_t valloc = fd_backtrace_alloc_virtual( &inner_valloc );
```

Comment:
don't think we should commit the backtrace in as default, make configurable somehow?

Reply:
removed